packages/panzer/adapters-stk/test/evaluator_tests/RandomFieldEvaluator.hpp

```diff
@@ -26,68 +27,40 @@ class PointEvaluation {
 
 template<typename EvalT, typename Traits>
 class RandomFieldEvaluator
-  :
-  public PHX::EvaluatorWithBaseImpl<Traits>,
-  public PHX::EvaluatorDerived<EvalT, Traits>
+  : public PHX::EvaluatorWithBaseImpl<Traits>,
+    public PHX::EvaluatorDerived<EvalT, Traits>
 {
   public:
+  void evaluateFields(typename Traits::EvalData d);
 
-    RandomFieldEvaluator(
-      const Teuchos::ParameterList& p);
-
-    void
-    postRegistrationSetup(
-      typename Traits::SetupData d,
-      PHX::FieldManager<Traits>& fm);
-
-    void
-    evaluateFields(
-      typename Traits::EvalData d);
-
-  private:
-
-    using ScalarT = typename EvalT::ScalarT;
-
-  PHX::MDField<ScalarT> field;
+private:
+  using ScalarT = typename EvalT::ScalarT;
+  PHX::MDField<ScalarT> field_;
+  Kokkos::Random_XorShift64_Pool<PHX::Device> random_pool_;
 
 public:
   RandomFieldEvaluator(const std::string & name,
                        const Teuchos::RCP<PHX::DataLayout> & dl)
-     : field(name,dl) { this->addEvaluatedField(field); }
-}; // end of class RandomFieldEvaluator
-
-
-//**********************************************************************
-template<typename EvalT, typename Traits>
-RandomFieldEvaluator<EvalT, Traits>::
-RandomFieldEvaluator(
-  const Teuchos::ParameterList& p)
-{
-   TEUCHOS_ASSERT(false); // don't do this
-}
-
-//**********************************************************************
-template<typename EvalT, typename Traits>
-void
-RandomFieldEvaluator<EvalT, Traits>::
-postRegistrationSetup(
-  typename Traits::SetupData  /* sd */,
-  PHX::FieldManager<Traits>&  fm)
-{
-  this->utils.setFieldData(field,fm);
-}
+    : field_(name,dl),random_pool_(12345)
+  { this->addEvaluatedField(field_); }
+};
 
 //**********************************************************************
 template<typename EvalT, typename Traits>
 void
 RandomFieldEvaluator<EvalT, Traits>::
-evaluateFields(
-  typename Traits::EvalData  /* workset */)
+evaluateFields(typename Traits::EvalData workset)
 {
-  auto field_h = Kokkos::create_mirror_view(field.get_view());
-   for(int i=0;i<static_cast<int>(field.size());i++)
-      field_h[i] = double(std::rand())/double(RAND_MAX);
-   Kokkos::deep_copy(field.get_view(),field_h);
+  auto field = field_;
+  auto random_pool = random_pool_;
+
+  Kokkos::parallel_for("panzer::RandomFieldEvalautor",workset.num_cells,KOKKOS_LAMBDA(const int i){
+    auto generator = random_pool.get_state();
+    for (size_t j=0; j < field.extent(1); ++j) {
+      field(i,j) = generator.drand(0.0,1.0);
+    }
+    random_pool.free_state(generator);
+  });
 }
 
 //**********************************************************************
```

Comment:
What is going on with these changes?

Reply:
This unit test was failing in the deep copy with sacado objects. The underlying issue was fixed and a unit test was added to sacado by Christian. A better fix for this evaluator was to just move the RNG computation to device. It was an old evaluator that was still doing computations on the host and deep copy back to device.